Admin/CrossTenantMailboxMigrationValidation.ps1

```diff
@@ -0,0 +1,480 @@
+.EXAMPLE
+
+        .\CrossTenantMailboxMigrationValidation.ps1 -SDP -LogPath C:\Temp\LogFile.txt
+
+        This will prompt you for the soureTenantId and TargetTenantId, establish 3 remote powershell sessions (one to the source EXO tenant, one to the target EXO tenant and another one to AAD target tenant), and will collect all the relevant information (config-wise) so it can be used for troubleshooting and send it to Microsoft Support if needed.
+.#>
+
+param (
+    [Parameter(Mandatory = $True, ParameterSetName = "ObjectsValidation", HelpMessage = "Validate source Mailbox and Target MailUser objects. If used alone you will be prompted to introduce the identities you want to validate")]
+    [System.Management.Automation.SwitchParameter]$CheckObjects,
+    [Parameter(Mandatory = $False, ParameterSetName = "ObjectsValidation", HelpMessage = "Path pointing to the CSV containing the identities to validate. CheckObjects parameter needs also to be specified")]
+    [System.String]$CSV,
+    [Parameter(Mandatory = $True, HelpMessage = "Path pointing to the log file")]
+    [System.String]$LogPath,
+    [Parameter(Mandatory = $True, ParameterSetName = "OrgsValidation", HelpMessage = "Validate the organizations settings like organization relationships, migraiton endpoint and AADApplication")]
+    [System.Management.Automation.SwitchParameter]$CheckOrgs,
+    [Parameter(Mandatory = $True, ParameterSetName = "SDP", HelpMessage = "Collect relevant data for troubleshooting purposes and send it to Microsoft Support if needed")]
+    [System.Management.Automation.SwitchParameter]$SDP
+)
+
+$wsh = New-Object -ComObject Wscript.Shell
+
+function ConnectToEXOTenants {
+    #Connect to SourceTenant (EXO)
+    Write-Verbose -Message "Informational: Connecting to SOURCE EXO tenant"
+    $wsh.Popup("You're about to connect to source tenant (EXO), please provide the SOURCE tenant admin credentials", 0, "SOURCE tenant") | Out-Null
+    Connect-ExchangeOnline -Prefix Source -ShowBanner:$false
+
+    #Connect to TargetTenant (EXO)
+    Write-Verbose -Message "Informational: Connecting to TARGET EXO tenant"
+    $wsh.Popup("You're about to connect to target tenant (EXO), please provide the TARGET tenant admin credentials", 0, "TARGET tenant") | Out-Null
+    Connect-ExchangeOnline -Prefix Target -ShowBanner:$false
+}
+
+function CheckObjects {
+
+    Write-Host "Informational: Loading SOURCE object"$SourceIdentity
+    $SourceObject = Get-SourceMailbox $SourceIdentity -ErrorAction SilentlyContinue
+    Write-Host "Informational: Loading TARGET object"$TargetIdentity
+    $TargetObject = Get-TargetMailUser $TargetIdentity -ErrorAction SilentlyContinue
+
```

Comment:
Export the `$sourceObject` and `$targetObject` out to clixml as well.

Reply:
@dpaulson45  Trying to recall why we want to do this?

Reply:
The export is to have a reference to the object that we were working with since we can be modifying them.